consensus/consensus-types/src/block.rs

```diff
@@ -309,3 +311,17 @@ impl<'de, T: DeserializeOwned + Serialize> Deserialize<'de> for Block<T> {
         })
     }
 }
+
+impl<T> From<&Block<T>> for BlockMetadata {
+    fn from(block: &Block<T>) -> Self {
+        Self::new(
+            block.id(),
+            block.timestamp_usecs(),
+            block.quorum_cert().ledger_info().signatures().clone(),
+            // For nil block, we use 0x0 which is convention for nil address in move.
+            block
+                .author()
+                .unwrap_or_else(|| AccountAddress::new([0u8; ADDRESS_LENGTH])),
```

Comment:
should this only happen for genesis? should we assert that?

Reply:
I think this also happens for any empty block (which can happen either to build a 3-chain after reconfiguration or due to lack of public chain traffic).

Reply:
this only happens for NIL block(the block we voted upon timeout) but not for empty blocks after reconfiguration(which has to be real empty, the trick to use is to bypass executor and roll over the parent's result)

there're many types of empty blocks
- genesis block - this is a virtual block that will never be executed
- proposer proposes a block without payload
- NIL block which every proposer individually generate and vote for upon timeout(this is the case we don't have a proposer)
- blocks after reconfiguration

Reply:
I wonder if at some time we should have predefined authors for different outcomes...
0x1 => genesis
0x2 => empty payload
0x3 => timeout block
etc

that way there's a cleaner way to reason about this field and we don't have meaningless fields in a block